tests/test_for_utils.py

```diff
@@ -116,3 +119,40 @@ def test_show_versions(capsys):
     assert lines[7].startswith("Package Versions")
     assert lines[8].startswith("================")
     assert any(l.startswith("memote") for l in lines[9:])
+
+
+@pytest.fixture(scope="function")
+def mock_repo(tmpdir_factory):
+    """Mock repo with history: add, modify, do nothing, delete file."""
+    path = tmpdir_factory.mktemp("memote-mock-repo")
+    repo = git.Repo.init(path)
+    relname = "test_file.txt"
+    absname = os.path.join(path, relname)
+
+    for message in "Add file.", "Modify file.":
+        with open(absname, "w") as f:
+            f.write(message)
+        repo.index.add([relname])
+        repo.index.commit(message)
+
+    repo.index.add([relname])
```

Comment:
I'm surprised this creates a commit at all. Usually adding an unmodified file has no effect and git won't let you make a commit without any changes. Maybe create and add another unrelated file here?

Reply:
This is a documented feature of git, but as the [docs](https://git-scm.com/docs/git-commit) say, mostly used for scripting conversion from other SCMs. I use it when reconstructing a history containing only xml and yml exports of the multi-file proprietary model format we've been struggling with.

git commit --allow-empty -m "Initial commit intentionally left blank."